MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClientDefault.java

```diff
@@ -456,23 +456,23 @@ public int getProductPageSize() {
     }
 
     @Override
-    public CancellableTask getProductPage(int page, Callback<List<Product>> callback) {
-        return productService.getProductPage(page, callback);
+    public CancellableTask getProducts(int page, Callback<List<Product>> callback) {
+        return productService.getProducts(page, callback);
     }
 
     @Override
-    public Observable<List<Product>> getProductPage(int page) {
-        return productService.getProductPage(page);
+    public Observable<List<Product>> getProducts(int page) {
+        return productService.getProducts(page);
     }
 
     @Override
-    public CancellableTask getProductWithHandle(String handle, Callback<Product> callback) {
-        return productService.getProductWithHandle(handle, callback);
+    public CancellableTask getProductByHandle(String handle, Callback<Product> callback) {
```

Comment:
Well the issue I see with that is we have `getProduct(Long productId)` so the signature almost the same (different type of parameter only), and could confuse user what method is for